k2/csrc/fsa_algo_test.cu

```diff
@@ -376,4 +374,26 @@ TEST(FsaAlgo, ShortestPath) {
   }
 }
 
+TEST(FsaAlgo, Union) {
+  std::string s1 = R"(0 1 1 0.1
+    0 2 2 0.2
+    1 3 -1 0.3
+    2 3 -1 0.4
+    3
+  )";
+  std::string s2 = R"(0 1 -1 0.5
+    1
+  )";
+
+  for (auto &context : {GetCpuContext(), GetCudaContext()}) {
+    Fsa fsa1 = FsaFromString(s1);
+    Fsa fsa2 = FsaFromString(s2);
+    Fsa *fsa_array[] = {&fsa1, &fsa2};
+    FsaVec fsa_vec = CreateFsaVec(2, &fsa_array[0]);
+    fsa_vec = fsa_vec.To(context);
+    Fsa fsa = Union(fsa_vec);
+    K2_LOG(INFO) << FsaToString(fsa);
```

Comment:
Again, I **strongly** suggest we add random tests (especially empty case) for any algorithms, it really will take our too much time to dubug in the final python training scripts if there's any issue. You can use RandomFsa/RandomFsaVec to create Fsas.

Reply:
Will add it.